buildSrc/src/main/kotlin/com/saveourtool/osv4k/buildutils/kotlin-library.gradle.kts

```diff
@@ -0,0 +1,121 @@
+    if (project.name == "save-common") {
+        // additionally, save-common should be available for JS too
+        // fixme: shouldn't rely on hardcoded project name here
+        js(IR).browser()
+
+        // store yarn.lock in the root directory
+        rootProject.extensions.configure<org.jetbrains.kotlin.gradle.targets.js.yarn.YarnRootExtension> {
+            lockFileDirectory = rootProject.projectDir
+        }
+    }
+
+    if (hasProperty("disableRedundantTargets") && (property("disableRedundantTargets") as String?) != "false") {
+        // with this flag we exclude targets that are present on multiple OS to speed up build
+        val currentOs = DefaultNativePlatform.getCurrentOperatingSystem()
+        val redundantTarget: String? = when {
+            currentOs.isWindows -> "linuxX64"
+            currentOs.isMacOsX -> "linuxX64"
+            currentOs.isLinux -> null
+            else -> throw GradleException("Unknown operating system ${currentOs.name}")
+        }
+        tasks.matching { redundantTarget != null && it.name.contains(redundantTarget, ignoreCase = true) }
+            .configureEach {
+                logger.lifecycle("Disabling task :${project.name}:$name on host $currentOs")
+                enabled = false
+            }
+    }
+
+    /*
+     * Common structure for MPP libraries:
+     *            common
+     *              |
+     *            nonJs
+     *          /       \
+     *       native      JVM
+     *     /   |    \
+     * linux  mingw macos
+     */
+    sourceSets {
+        all {
+            languageSettings.optIn("kotlin.RequiresOptIn")
+        }
+        val commonMain by getting
+        val commonTest by getting {
+            dependencies {
+                implementation("io.kotest:kotest-assertions-core:5.6.2")
+            }
+        }
+        val commonNonJsMain by creating {
+            dependsOn(commonMain)
+        }
+        val commonNonJsTest by creating {
+            dependsOn(commonTest)
+            dependencies {
+                implementation(kotlin("test-common"))
+                implementation(kotlin("test-annotations-common"))
+            }
+        }
+        val jvmMain by getting {
+            dependsOn(commonNonJsMain)
+        }
+        val jvmTest by getting {
+            dependsOn(commonNonJsTest)
+            dependencies {
+                implementation(kotlin("test-junit5"))
+                implementation("org.junit.jupiter:junit-jupiter-engine:5.10.0")
+            }
+        }
+        val nativeMain by creating {
+            dependsOn(commonNonJsMain)
+        }
+        val nativeTest by creating {
+            dependsOn(commonNonJsTest)
+        }
+        nativeTargets.forEach {
+            getByName("${it.name}Main").dependsOn(nativeMain)
+        }
+        nativeTargets.forEach {
+            getByName("${it.name}Test").dependsOn(nativeTest)
+        }
+    }
+}
+
+// configureJacoco()
+// configurePublishing()
```

Comment:
why not now? relkeases could be done at any time later

Reply:
working on it right now. I had problem with gradle and finally resolved it

Reply:
fixed